### PR TITLE
Ensure default values are passed to nested installers within zip

### LINKED ISSUE #2392: Zip installers not getting nested installer type defaults
### Brief description of your issue

Installers within zip files are not getting the default values for their nested installer, such as switches and error codes.

### Steps to reproduce

Use zip.

### Expected behavior

Works as if not zipped.

### Actual behavior

Has no command line arguments passed.

### Environment

```shell
CI/CD
```

### PR DESCRIPTION
**Pull request by ryfu-msft**
Fixes #2392 

Changes:
- Changes installer.InstallerType -> installer.BaseInstallerType.
- Created a method called EffectiveInstallerType() that returns the actual installerType that will be executed (i.e. the NestedInstallerType if it is a zip installer)

Tests:
- Reenabled ZipWithMsi test
- All existing tests should pass

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/2413)

### CODE REVIEW

File: src/AppInstallerCLICore/Workflows/DownloadFlow.cpp [hunk 1]
```diff
@@ -183,7 +183,7 @@ namespace AppInstaller::CLI::Workflow
     void CheckForExistingInstaller(Execution::Context& context)
     {
         const auto& installer = context.Get<Execution::Data::Installer>().value();
-        if (installer.InstallerType == InstallerTypeEnum::MSStore)
+        if (installer.EffectiveInstallerType() == InstallerTypeEnum::MSStore)
```
Comment (JohnMcPMS) [inline src/AppInstallerCLICore/Workflows/DownloadFlow.cpp:186]: [This is the correct usage here just calling this out if it isn't already done...]
As `MSStore` is not an installer type that actually has an installer file (to us), it should not be a valid nested type.  Is that already handled elsewhere?

File: src/AppInstallerCLICore/Workflows/ShowFlow.cpp [hunk 2]
```diff
@@ -173,7 +173,7 @@ namespace AppInstaller::CLI::Workflow
         info << Execution::ManifestInfoEmphasis << Resource::String::ShowLabelInstaller << std::endl;
         if (installer)
         {
-            info << "  "_liv << Execution::ManifestInfoEmphasis << Resource::String::ShowLabelInstallerType << ' ' << Manifest::InstallerTypeToString(installer->InstallerType) << std::endl;
+            info << "  "_liv << Execution::ManifestInfoEmphasis << Resource::String::ShowLabelInstallerType << ' ' << Manifest::InstallerTypeToString(installer->BaseInstallerType) << std::endl;
```
Comment (JohnMcPMS) [inline src/AppInstallerCLICore/Workflows/ShowFlow.cpp:176]: I think that this should be something like:
```
Effective (Base)
```
When effective and base are not the same.  The important part is the effective type, but also showing the base is useful.

File: src/AppInstallerRepositoryCore/Rest/Schema/1_1/Json/ManifestDeserializer_1_1.cpp [hunk 3]
```diff
@@ -158,7 +158,7 @@ namespace AppInstaller::Repository::Rest::Schema::V1_1::Json
             }
 
             // Populate installer default return codes if not present in ExpectedReturnCodes and InstallerSuccessCodes
-            auto defaultReturnCodes = GetDefaultKnownReturnCodes(installer.InstallerType);
+            auto defaultReturnCodes = GetDefaultKnownReturnCodes(installer.BaseInstallerType);
```
Comment (JohnMcPMS) [inline src/AppInstallerRepositoryCore/Rest/Schema/1_1/Json/ManifestDeserializer_1_1.cpp:161]: Effective, unless it is not yet know, then some more effort will be required.

File: src/AppInstallerRepositoryCore/Rest/Schema/1_0/Json/ManifestDeserializer_1_0.cpp [hunk 4]
```diff
@@ -366,7 +366,7 @@ namespace AppInstaller::Repository::Rest::Schema::V1_0::Json
         }
 
         // Installer Switches
-        installer.Switches = Manifest::GetDefaultKnownSwitches(installer.InstallerType);
+        installer.Switches = Manifest::GetDefaultKnownSwitches(installer.BaseInstallerType);
```
Comment (JohnMcPMS) [inline src/AppInstallerRepositoryCore/Rest/Schema/1_0/Json/ManifestDeserializer_1_0.cpp:369]: Effective, unless it is not yet know, then some more effort will be required.

**Review by JohnMcPMS — CHANGES_REQUESTED** [review]

File: src/AppInstallerRepositoryCore/Rest/Schema/1_0/Json/ManifestDeserializer_1_0.cpp (hunk 4 above)
Comment (ryfu-msft) [reply src/AppInstallerRepositoryCore/Rest/Schema/1_0/Json/ManifestDeserializer_1_0.cpp]: Changed to effective

File: src/AppInstallerRepositoryCore/Rest/Schema/1_1/Json/ManifestDeserializer_1_1.cpp (hunk 3 above)
Comment (ryfu-msft) [reply src/AppInstallerRepositoryCore/Rest/Schema/1_1/Json/ManifestDeserializer_1_1.cpp]: Changed to effective

File: src/AppInstallerCLICore/Workflows/ShowFlow.cpp (hunk 2 above)
Comment (ryfu-msft) [reply src/AppInstallerCLICore/Workflows/ShowFlow.cpp]: Changed to suggested. 
 
If effective and base are not the same, will show both "Effective (Base)". Added tests to verify output

File: src/AppInstallerCLICore/Workflows/DownloadFlow.cpp (hunk 1 above)
Comment (ryfu-msft) [reply src/AppInstallerCLICore/Workflows/DownloadFlow.cpp:186]: The only check I had was from manifest validation. I added another check prior to the install flow to verify that if the NestedInstallertype is valid if the BaseInstallertype is an archive. Outputs the appropriate error message if NestedInstallertype is not supported.

**Review by JohnMcPMS — APPROVED** [review]

**Pull request merged**